packages/relic/lib/src/headers/headers.dart

```diff
@@ -0,0 +1,1688 @@
+    _retryAfterHeader,
+    _trailerHeader,
+    _transferEncodingHeader,
+    _varyHeader,
+    _wwwAuthenticateHeader,
+    _xPoweredByHeader,
+
+    // Common Headers (Used in Both Requests and Responses)
+    _acceptHeader,
+    _acceptRangesHeader,
+    _contentLengthHeader,
+    _contentTypeHeader,
+    _cookieHeader,
+    _setCookieHeader,
+
+    // Security and Modern Headers
+    _accessControlAllowHeadersHeader,
+    _accessControlAllowMethodsHeader,
+    _clearSiteDataHeader,
+    _contentSecurityPolicyHeader,
+    _crossOriginEmbedderPolicyHeader,
+    _crossOriginOpenerPolicyHeader,
+    _crossOriginResourcePolicyHeader,
+    _permissionsPolicyHeader,
+    _referrerPolicyHeader,
+    _secFetchDestHeader,
+    _secFetchModeHeader,
+    _secFetchSiteHeader,
+    _strictTransportSecurityHeader,
+  };
+
+  Headers._({
+    // Date-related headers
+    this.date,
+    this.expires,
+    this.ifModifiedSince,
+    this.lastModified,
+
+    // General Headers
+    this.origin,
+    this.server,
+    this.via,
+
+    // Request Headers
+    this.from,
+    this.host,
+    this.acceptEncoding,
+    this.acceptLanguage,
+    this.accessControlRequestHeaders,
+    this.accessControlRequestMethod,
+    this.age,
+    this.allow,
+    this.authorization,
+    this.connection,
+    this.expect,
+    this.ifMatch,
+    this.ifNoneMatch,
+    this.ifRange,
+    this.maxForwards,
+    this.proxyAuthorization,
+    this.range,
+    this.referer,
+    this.userAgent,
+    this.te,
+    this.upgrade,
+
+    // Response Headers
+    this.location,
+    this.xPoweredBy,
+    this.accessControlAllowCredentials,
+    this.accessControlAllowOrigin,
+    this.accessControlExposeHeaders,
+    this.accessControlMaxAge,
+    this.cacheControl,
+    this.contentEncoding,
+    this.contentLanguage,
+    this.contentLocation,
+    this.contentRange,
+    this.etag,
+    this.proxyAuthenticate,
+    this.retryAfter,
+    this.trailer,
+    this.vary,
+    this.wwwAuthenticate,
+    this.contentDisposition,
+
+    // Common Headers (Used in Both Requests and Responses)
+    this.accept,
+    this.acceptRanges,
+    this.transferEncoding,
+    this.cookie,
+    this.setCookie,
+    CustomHeaders? custom,
+
+    // Security and Modern Headers
+    this.strictTransportSecurity,
+    this.contentSecurityPolicy,
+    this.referrerPolicy,
+    this.permissionsPolicy,
+    this.accessControlAllowMethods,
+    this.accessControlAllowHeaders,
+    this.clearSiteData,
+    this.secFetchDest,
+    this.secFetchMode,
+    this.secFetchSite,
+    this.crossOriginResourcePolicy,
+    this.crossOriginEmbedderPolicy,
+    this.crossOriginOpenerPolicy,
+    required this.failedHeadersToParse,
+  }) : custom = custom ?? CustomHeaders.empty();
+
+  factory Headers.fromHttpRequest(
+    io.HttpRequest request, {
+    bool strict = false,
+    required String? xPoweredBy,
+  }) {
+    Map<String, List<String>> failedHeadersToParse = {};
+    var dartIOHeaders = request.getHeadersParser(
```

Comment:
Suggestion: Remove extension and create the HeaderParser directly.

This is only called from this single place. No need to add an extention to request to support this.